local WeaponCatalog = {
	{
		Id = "crowbar",
		DisplayName = "Crowbar",
		TemplateName = "Crowbar",
		-- Weapon is visible/usable only for classes whose WeaponTag matches this ClassTag.
		ClassTag = "Berserker",
		Cost = 100,
		WeaponClass = "Melee",
		Damage = 20,
		Range = 8,
		Cooldown = 0.65,
		SwingDuration = 0.24,
		SwingForwardDegrees = 76,
		SwingSideDegrees = 24,
		SwingRollDegrees = -14,
		SwingWindupDegrees = 64,
		SwingWindupSideDegrees = -34,
		SwingWindupRollDegrees = 24,
		MeleeDownwardBias = 0.42,
		MeleeOriginDrop = 1.35,
		MeleeGroundRangeBonus = 1.35,
		MeleeGroundArcBonus = 18,
		MeleeGroundExtraRadius = 2.2,
		ArcDegrees = 90,
		Color = Color3.fromRGB(115, 115, 120),
		GripPositionOffset = Vector3.new(0.7, -0.6, 0),
		GripRotationDeg = Vector3.new(0, 90, 0),
	},
	{
		Id = "Linked Sword",
		DisplayName = "Linked Sword",
		TemplateName = "Linked Sword",
		-- Weapon is visible/usable only for classes whose WeaponTag matches this ClassTag.
		ClassTag = "Berserker",
		Cost = 100,
		WeaponClass = "Melee",
		Damage = 20,
		Range = 8,
		Cooldown = 0.65,
		SwingDuration = 0.24,
		SwingForwardDegrees = 76,
		SwingSideDegrees = 24,
		SwingRollDegrees = -14,
		SwingWindupDegrees = 64,
		SwingWindupSideDegrees = -34,
		SwingWindupRollDegrees = 24,
		MeleeDownwardBias = 0.42,
		MeleeOriginDrop = 1.35,
		MeleeGroundRangeBonus = 1.35,
		MeleeGroundArcBonus = 18,
		MeleeGroundExtraRadius = 2.2,
		ArcDegrees = 90,
		Color = Color3.fromRGB(115, 115, 120),
		GripPositionOffset = Vector3.new(0.7, -0.6, 0),
		GripRotationDeg = Vector3.new(0, 90, 0),
	},
	{
		Id = "Axe",
		DisplayName = "Axe",
		TemplateName = "Axe",
		-- Weapon is visible/usable only for classes whose WeaponTag matches this ClassTag.
		ClassTag = "Berserker",
		Cost = 100,
		WeaponClass = "Melee",
		Damage = 20,
		Range = 8,
		Cooldown = 0.65,
		SwingDuration = 0.24,
		SwingForwardDegrees = 76,
		SwingSideDegrees = 24,
		SwingRollDegrees = -14,
		SwingWindupDegrees = 64,
		SwingWindupSideDegrees = -34,
		SwingWindupRollDegrees = 24,
		MeleeDownwardBias = 0.42,
		MeleeOriginDrop = 1.35,
		MeleeGroundRangeBonus = 1.35,
		MeleeGroundArcBonus = 18,
		MeleeGroundExtraRadius = 2.2,
		ArcDegrees = 90,
		Color = Color3.fromRGB(115, 115, 120),
		GripPositionOffset = Vector3.new(0.1, -0.1, 0),
		GripRotationDeg = Vector3.new(0, 270, 0),
	},
	{
		Id = "CHAINSAW",
		DisplayName = "CHAINSAW",
		TemplateName = "CHAINSAW",
		-- Weapon is visible/usable only for classes whose WeaponTag matches this ClassTag.
		ClassTag = "Berserker",
		Cost = 100,
		WeaponClass = "Melee",
		Damage = 20,
		Range = 8,
		Cooldown = 0.65,
		SwingDuration = 0.24,
		SwingForwardDegrees = 76,
		SwingSideDegrees = 84,
		SwingRollDegrees = -64,
		SwingWindupDegrees = 24,
		SwingWindupSideDegrees = -34,
		SwingWindupRollDegrees = 24,
		MeleeDownwardBias = 0.42,
		MeleeOriginDrop = 1.35,
		MeleeGroundRangeBonus = 1.35,
		MeleeGroundArcBonus = 18,
		MeleeGroundExtraRadius = 2.2,
		ArcDegrees = 90,
		Color = Color3.fromRGB(115, 115, 120),
		GripPositionOffset = Vector3.new(0.3, 0.7, 1.5),
		GripRotationDeg = Vector3.new(0, 0, 34),
	},
	{
        Id = "Scythe",
        DisplayName = "Scythe",
        TemplateName = "Scythe",
        -- Weapon is visible/usable only for classes whose WeaponTag matches this ClassTag.
        ClassTag = "Berserker",
        Cost = 100,
        WeaponClass = "Melee",
        Damage = 20,
        Range = 8,
        Cooldown = 0.65,
        SwingDuration = 0.24,
        SwingForwardDegrees = 76,
        SwingSideDegrees = 84,
        SwingRollDegrees = -64,
        SwingWindupDegrees = 24,
        SwingWindupSideDegrees = -34,
        SwingWindupRollDegrees = 24,
        MeleeDownwardBias = 0.42,
        MeleeOriginDrop = 1.35,
        MeleeGroundRangeBonus = 1.35,
        MeleeGroundArcBonus = 18,
        MeleeGroundExtraRadius = 2.2,
        ArcDegrees = 90,
        Color = Color3.fromRGB(0, 0, 0),
        GripPositionOffset = Vector3.new(-0.7, 0.8, -1.8),
        GripRotationDeg = Vector3.new(105, 177, 0),
    }, 
	{
		Id = "Anti-Magic Sword",
		DisplayName = "Anti-Magic Sword",
		TemplateName = "Anti-Magic Sword",
		-- Weapon is visible/usable only for classes whose WeaponTag matches this ClassTag.
		ClassTag = "Berserker",
		Cost = 100,
		WeaponClass = "Melee",
		Damage = 20,
		Range = 8,
		Cooldown = 0.65,
		SwingDuration = 0.24,
		SwingForwardDegrees = 76,
		SwingSideDegrees = 24,
		SwingRollDegrees = -14,
		SwingWindupDegrees = 64,
		SwingWindupSideDegrees = -34,
		SwingWindupRollDegrees = 24,
		MeleeDownwardBias = 0.42,
		MeleeOriginDrop = 1.35,
		MeleeGroundRangeBonus = 1.35,
		MeleeGroundArcBonus = 18,
		MeleeGroundExtraRadius = 2.2,
		ArcDegrees = 90,
		Color = Color3.fromRGB(115, 115, 120),
		GripPositionOffset = Vector3.new(4.7, 0.7, 0),
		GripRotationDeg = Vector3.new(110, 0, 80),
	},
	{
		Id = "Black Sun Katana",
		DisplayName = "Black Sun Katana",
		TemplateName = "Black Sun Katana",
		-- Weapon is visible/usable only for classes whose WeaponTag matches this ClassTag.
		ClassTag = "Berserker",
		Cost = 100,
		WeaponClass = "Melee",
		Damage = 20,
		Range = 8,
		Cooldown = 0.65,
		SwingDuration = 0.24,
		SwingForwardDegrees = 76,
		SwingSideDegrees = 24,
		SwingRollDegrees = -14,
		SwingWindupDegrees = 64,
		SwingWindupSideDegrees = -34,
		SwingWindupRollDegrees = 24,
		MeleeDownwardBias = 0.42,
		MeleeOriginDrop = 1.35,
		MeleeGroundRangeBonus = 1.35,
		MeleeGroundArcBonus = 18,
		MeleeGroundExtraRadius = 2.2,
		ArcDegrees = 90,
		Color = Color3.fromRGB(115, 115, 120),
		GripPositionOffset = Vector3.new(0.4, 0.9, -1.7),
		GripRotationDeg = Vector3.new(0, 60, 90),
	},
	{
		Id = "rifle",
		DisplayName = "Rifle",
		ClassTag = "Sniper",
		Cost = 900,
		WeaponClass = "Ranged",
		Damage = 145,
		Range = 150,
		Cooldown = 1.45,
		MagazineSize = 5,
		ReserveAmmo = 30,
		ReloadSeconds = 2.9,
		ProjectileSpeed = 300,
		SpreadDegrees = 0.45,
		Color = Color3.fromRGB(90, 140, 95),
	},
	{
		Id = "awp",
		DisplayName = "AWP",
		TemplateName = "awp",
		ClassTag = "Sniper",
		Cost = 2200,
		WeaponClass = "Ranged",
		Damage = 165,
		Range = 175,
		Cooldown = 1.55,
		MagazineSize = 6,
		ReserveAmmo = 36,
		ReloadSeconds = 3.0,
		ProjectileSpeed = 320,
		SpreadDegrees = 0.22,
		Scoped = true,
		AutoScopeInFirstPerson = true,
		ScopeZoomFov = 18,
		ScopeOverlayStyle = "AWP",
		Color = Color3.fromRGB(78, 126, 86),
		GripPositionOffset = Vector3.new(0.9, 0.75, 0.05),
		GripRotationDeg = Vector3.new(0, 270, 0),
	},
	{
		Id = "railgun",
		DisplayName = "Railgun",
		TemplateName = "railgun",
		ClassTag = "Engineer",
		Cost = 1800,
		WeaponClass = "Ranged",
		Damage = 80,
		Range = 90,
		Cooldown = 0.9,
		MagazineSize = 6,
		ReserveAmmo = 30,
		ReloadSeconds = 2.8,
		ProjectileSpeed = 420,
		Color = Color3.fromRGB(120, 170, 220),
		GripRotationDeg = Vector3.new(0, 270, 0),
	},
	{
		Id = "flamethrower",
		DisplayName = "Flamethrower",
		ClassId = "pyromaniac",
		ClassTag = "Pyro",
		Cost = 2500,
		WeaponClass = "Flamethrower",
		Damage = 11,
		Range = 30,
		Cooldown = 0.12,
		MagazineSize = 50,
		ReserveAmmo = 320,
		ReloadSeconds = 3.5,
		ArcDegrees = 60,
		BurstTicks = 6,
		BurstIntervalSeconds = 0.07,
		AutoFire = true,
		GroundFireChance = 0.08,
		GroundFireDamageScale = 0.1,
		GroundFireRadius = 1,
		GroundFireDurationSeconds = 2.8,
		GroundFireTickIntervalSeconds = 0.35,
		GroundFireSpawnCooldownSeconds = 0.5,
		Color = Color3.fromRGB(240, 145, 55),
	},
	{
		Id = "flare_gun",
		DisplayName = "Flare Gun",
		TemplateName = "flare gun",
		ClassId = "pyromaniac",
		ClassTag = "Pyro",
		Cost = 1800,
		WeaponClass = "FlareGun",
		Damage = 60,
		Range = 95,
		Cooldown = 0.7,
		MagazineSize = 1,
		ReserveAmmo = 18,
		ReloadSeconds = 2.2,
		ProjectileSpeed = 95,
		ProjectileGravityScale = 0.08,
		BlastRadius = 16,
		BurnDurationSeconds = 4,
		BurnTickIntervalSeconds = 0.5,
		BurnDamageFractionPerSecond = 0.18,
		BurnBaseDamage = 60,
		Color = Color3.fromRGB(255, 105, 70),
	},
	{
		Id = "grenade_launcher",
		DisplayName = "Grenade Launcher",
		TemplateName = "m32a1 grenade launcher",
		ClassTag = "Demolitionist",
		Cost = 2850,
		WeaponClass = "GrenadeLauncher",
		Damage = 95,
		Range = 125,
		Cooldown = 1.05,
		MagazineSize = 4,
		ReserveAmmo = 20,
		ReloadSeconds = 3.2,
		ProjectileSpeed = 110,
		BlastRadius = 14,
		FuseSeconds = 2.8,
		LobFactor = 0.18,
		Color = Color3.fromRGB(108, 122, 82),
		GripPositionOffset = Vector3.new(-0.1, 0.3, -0.65),
		GripRotationDeg = Vector3.new(0, 270, 0),
	},
	{
		Id = "rpg",
		DisplayName = "RPG",
		ClassTag = "Demolitionist",
		Cost = 3200,
		WeaponClass = "RPG",
		Damage = 120,
		Range = 140,
		Cooldown = 0.95,
		MagazineSize = 1,
		ReserveAmmo = 8,
		ReloadSeconds = 2.8,
		ProjectileSpeed = 145,
		BlastRadius = 18,
		Color = Color3.fromRGB(110, 125, 80),
	},
	{
		Id = "ak",
		DisplayName = "AK-47",
		TemplateName = "AK-47",
		ClassTag = "Commander",
		Cost = 1600,
		WeaponClass = "Ranged",
		Damage = 36,
		Range = 78,
		Cooldown = 0.1,
		MagazineSize = 30,
		ReserveAmmo = 180,
		ReloadSeconds = 2.45,
		ProjectileSpeed = 340,
		SpreadDegrees = 3.2,
		AutoFire = true,
		Color = Color3.fromRGB(96, 122, 88),
		GripPositionOffset = Vector3.new(1, 0.8, 0.1),
		GripRotationDeg = Vector3.new(0, 90, 0),
	},
	{
		Id = "sentry_lv1",
		DisplayName = "Sentry Lv1",
		TemplateName = "level 1 sentry",
		ClassTag = "SentryTech",
		Cost = 1300,
		WeaponClass = "SentryDeployer",
		Damage = 0,
		Range = 180,
		Cooldown = 0.7,
		TurretMaxHealth = 420,
		TurretMaxAmmo = 300,
		TurretDamage = 10,
		TurretFireCooldown = 0.5,
		TurretAcquireRange = 100000,
		TurretThinkInterval = 0.08,
		TurretPlacementDistance = 30,
		Color = Color3.fromRGB(110, 120, 138),
	},
	{
		Id = "mimi_sentry",
		DisplayName = "Mimi Sentry",
		TemplateName = "Mimi Sentry",
		ClassTag = "SentryTech",
		Cost = 1300,
		WeaponClass = "SentryDeployer",
		Damage = 0,
		Range = 180,
		Cooldown = 0.7,
		TurretMaxHealth = 420,
		TurretMaxAmmo = 300,
		TurretDamage = 10,
		TurretFireCooldown = 0.5,
		TurretAcquireRange = 100000,
		TurretThinkInterval = 0.08,
		TurretPlacementDistance = 30,
		TurretAttackMode = "RocketBurst",
		TurretRocketCount = 2,
		TurretRocketBlastRadius = 8,
		TurretRocketProjectileSpeed = 145,
		TurretRocketRange = 180,
		TurretRocketSpreadDegrees = 4,
		TurretMuzzleOffset = Vector3.new(0.8, 0, 4),
		TurretPlacementRotationDeg = Vector3.new(0, 160, 0),
		Color = Color3.fromRGB(210, 120, 190),
	},
	{
		Id = "repair_tool",
		DisplayName = "Repair Tool",
		TemplateName = "repair tool",
		ClassTag = "SentryTech",
		Cost = 950,
		WeaponClass = "RepairTool",
		Damage = 0,
		Range = 80,
		Cooldown = 0.18,
		AutoFire = true,
		RepairPerSecond = 24,
		RepairDistance = 12,
		Color = Color3.fromRGB(102, 186, 205),
	},
}

return WeaponCatalog
