-- ServerScriptService.Features.Enemies exports
return {
	EnemyService = require(script.EnemyService),
	EnemyAIService = require(script.EnemyAIService),
	EnemyFactory = require(script.EnemyFactory),
	EnemyRegistry = require(script.EnemyRegistry),
	EnemyScaling = require(script.EnemyScaling),
	EnemySpawnResolver = require(script.EnemySpawnResolver),
	EnemyTargeting = require(script.EnemyTargeting),
	EnemyVfxService = require(script.EnemyVfxService),
	ModelRootResolver = require(script.ModelRootResolver),
}
